ParkingZoneApp/Services/IParkingSlotService.cs

```diff
@@ -1,11 +1,12 @@
 ﻿using ParkingZoneApp.Models;
-using ParkingZoneApp.Repositories;
 
 namespace ParkingZoneApp.Services
 {
     public interface IParkingSlotService : IService<ParkingSlot>
     {
         public IEnumerable<ParkingSlot> GetByParkingZoneId(int parkingZoneId);
-        public bool ParkingSlotExits(int ParkingZoneId, int ParkingSlotNumber);
+        public bool ParkingSlotExists(int parkingZoneId, int parkingSlotNumber);
+        public bool SlotFree(ParkingSlot slot, DateTime starTime, int duration);
```

Comment:
when we read method name, we should understand what it does. in your case its little bit weird. (naming convention)